test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js

```diff
@@ -26,6 +26,94 @@ const host = dataHelper.getJetpackHost();
 
 const siteEditorUser = 'siteEditorSimpleSiteUser';
 
+const exitSiteEditor = async function ( driver ) {
+	const isSiteEditorOpen = await driverHelper.isElementLocated(
+		driver,
+		By.css( '.edit-site-header' )
+	);
+	if ( ! isSiteEditorOpen ) {
+		return;
+	}
+
+	const editor = await SiteEditorComponent.Expect( driver );
+	const isNavigationSidebarOpen = await driverHelper.isElementLocated(
+		driver,
+		By.css( '.edit-site-navigation-panel.is-open' )
+	);
+	if ( ! isNavigationSidebarOpen ) {
+		await editor.toggleNavigationSidebar();
+	}
+
+	await driverHelper.waitUntilElementLocatedAndVisible(
+		driver,
+		By.css( '.components-navigation__back-button, .edit-site-navigation-panel__back-to-dashboard' )
+	);
+
+	while (
+		! ( await driverHelper.isElementLocated(
+			driver,
+			By.css( '.edit-site-navigation-panel__back-to-dashboard' )
+		) )
+	) {
+		await driverHelper.clickWhenClickable(
+			driver,
+			By.css( '.components-navigation__back-button' )
+		);
+	}
```

Comment:
We should be able to use the native `wait` loop here:
```suggestion
	await this.driver.wait(
		async function ( driver ) {
			if (
				await driverHelper.isElementNotLocated(
					driver,
					By.css( '.edit-site-navigation-panel__back-to-dashboard' )
				)
			) {
				await driverHelper.clickWhenClickable(
					driver,
					By.css( '.components-navigation__back-button' )
				);
				return false;
			}
			return true;
		},
		config.get( 'explicitWaitMS' ),
		'Could not reach the "Dashboard" button'
	);
```

This makes sure [the condition is polled every 200ms](https://github.com/SeleniumHQ/selenium/blob/68f3edcb9b28925757d3f19ba2a255d7d2d6b5b6/javascript/node/selenium-webdriver/lib/webdriver.js#L807) which is safer comparing to the `while` loop that will bomb the document with element queries. It will also timeout at the correct moment, providing a useful error message.